instrumentation/src/main.c

```diff
@@ -0,0 +1,99 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#define MAX_LINE_LENGTH 1023
+#define MAX_LINES 50
+
+#define ALLOWED_ENV_VARS "OTEL_SERVICE_NAME", "OTEL_EXPORTER_OTLP_ENDPOINT", "OTEL_RESOURCE_ATTRIBUTES", "SPLUNK_PROFILER_ENABLED", "SPLUNK_PROFILER_MEMORY_ENABLED", "SPLUNK_METRICS_ENABLED", "JAVA_TOOL_OPTIONS", "NODE_OPTIONS"
+
+static char *const allowed_env_vars[] = {ALLOWED_ENV_VARS};
+static size_t const allowed_env_vars_size = sizeof(allowed_env_vars) / sizeof(*allowed_env_vars);
+
+// TODO change to systemd drop in file paths
```

Comment:
If using the unit files I think we'll want to use a vetted ini? I'm not sure this would be a desirable conf form though since they are ultimately for configuring different things.

Reply:
Sure, I'll look into that.